datalad/cmd.py

```diff
@@ -418,6 +416,72 @@ def run(self, cmd, protocol=None, stdin=None, cwd=None, env=None, **kwargs):
         results.pop('code', None)
         return results
 
+    @classmethod
+    def _check_if_new_proc(cls):
+        """Check if WitlessRunner is used under a new PID
+
+        Raises
+        ------
+        RuntimeError
+          If it is not a new proc and we already know that we need a new loop
+          in this pid
+        """
+        pid = os.getpid()
+        is_new_proc = cls._loop_pid is None or cls._loop_pid != pid
+        if is_new_proc:
+            # We need to check if we can run any command smoothly
+            cls._loop_pid = pid
+            cls._loop_need_new = None
+        elif cls._loop_need_new:
+            raise RuntimeError("we know we need a new loop")
+        return is_new_proc
+
+    @classmethod
+    def _check_if_loop_usable(cls, event_loop, stdin):
+        """Check if given event_loop could run a simple command
+
+        Sets _loop_need_new variable to a bool depending on what it finds
+
+        Raises
+        ------
+        RuntimeError
+          If loop is not reusable
+        """
+        # We need to check if we can run any command smoothly
+        try:
+            event_loop.run_until_complete(
+                run_async_cmd(
+                    event_loop,
+                    [sys.executable, "--version"],  # fast! 0.004 sec and to be ran once per process
+                    KillOutput,
+                    stdin,
+                )
+            )
+            cls._loop_need_new = False
+        except OSError as e:
+            # due to https://bugs.python.org/issue21998
+            # exhibits in https://github.com/ReproNim/testkraken/issues/95
+            lgr.debug("It seems we need a new loop when running our commands: %s", exc_str(e))
+            cls._loop_need_new = True
+            raise RuntimeError("the loop is not reusable")
```

Comment:
Same spirit as the comment above: Why is this a runtime error? Why not a new loop?

Reply:
see above, and it will be a new loop since we are raising an exception here.